module/VuFind/src/VuFind/View/Helper/Root/Matomo.php

```diff
@@ -489,14 +489,11 @@ protected function getPageUrl(): string
         $path = $this->request->getUri()->toString();
         $routeMatch = $this->router->match($this->request);
         if ($routeMatch
-            && $routeMatch->getMatchedRouteName() == 'vufindrecord-ajaxtab'
+            && substr($routeMatch->getMatchedRouteName(), -8) === '-ajaxtab'
+            && null !== ($pos = strrpos($path, '/AjaxTab'))
+            && ($tab = $this->request->getPost('tab'))
         ) {
-            // Replace 'AjaxTab' with tab name in record page URLs
```

Comment:
Should this comment be retained to clarify what is happening here? It's a bit dense to work through otherwise.

Reply:
Yes. I've now put it back but above the whole block.